### PR TITLE
only detect advertise address if derived value is any

### PR DESCRIPTION
**Pull request by magiconair**

### CODE REVIEW

File: agent/config/builder.go [hunk 1]
```diff
@@ -342,8 +342,8 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 		}
 	}
 
-	advertiseAddr := bindAddr
-	if detect != nil {
+	advertiseAddr := b.makeIPAddr(b.expandFirstIP("advertise_addr", c.AdvertiseAddrLAN), bindAddr)
+	if ipaddr.IsAny(advertiseAddr) {
```
Comment (slackpad) [inline agent/config/builder.go]: I think you can end up in here with advertise set to `0.0.0.0` and bind set to non-ANY with `detect` set to `nil`, so we probably still need a `nil` check.

**Comment by magiconair** [comment]
You're right. Pls have a look at the fix.

File: agent/config/builder.go (hunk 1 above)
Comment (magiconair) [reply agent/config/builder.go]: fixed

**Comment by slackpad** [comment]
LGTM - merged!

**Pull request merged**